### PR TITLE
Update policy assignments and exemptions api-version to support selectors and overrides

### PR DESCRIPTION
**Pull request by robga**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month. 2022-09-20
  3. When do you expect to publish the swagger? Please provide date or, the date is not yet available, the month. 2022-10
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[x] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by robga** [comment]
@ArcturusZhang Could you take a look at the PR? I don't think the failed checks are related to my changes. They seem complying about the existing files in the repo.

File: specification/resources/resource-manager/readme.go.md [hunk 1]
```diff
@@ -208,6 +209,16 @@ namespace: locks
...
+### Tag: package-policy-2022-06 and go
+
+These settings apply only when `--tag=package-policy-2022-06 --go` is specified on the command line.
+Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.
+
+``` yaml $(tag) == 'package-policy-2022-06' && $(go)
+namespace: policy
+output-folder: $(go-sdk-folder)/services/preview/resources/mgmt/2022-06-01-preview/policy
+```
+
```
Comment (ArcturusZhang) [inline specification/resources/resource-manager/readme.go.md:221]: These configurations are for Go Track 1 SDK which will be deprecated very soon, therefore we do not need to update this file.
Could you please revert the changes here?

**Comment by robga** [comment]
@leni-msft Could you take a look at the PR? I don't think the failed checks are related to my changes. They seem complying about the existing files in the repo.

File: specification/resources/resource-manager/Microsoft.Authorization/preview/2022-07-01-preview/policyExemptions.json [hunk 2]
```diff
@@ -0,0 +1,699 @@
...
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/PolicyExemptionProperties",
+          "description": "Properties for the policy exemption."
+        },
+        "systemData": {
+          "readOnly": true,
+          "type": "object",
+          "description": "Azure Resource Manager metadata containing createdBy and modifiedBy information.",
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
```
Comment (leni-msft) [inline specification/resources/resource-manager/Microsoft.Authorization/preview/2022-07-01-preview/policyExemptions.json:549]: new apiVersion should reference v3/types.json

**Comment by leni-msft** [comment]
@jianyexi please help check lint exception

**Comment by leni-msft** [comment]
@robga To fix Avocado error, please make sure all APIs are included in the new default tag

**Comment by robga** [comment]
> @robga To fix Avocado error, please make sure all APIs are included in the new default tag

@leni-msft  I think that is a false positive error. All the APIs have been included in the new tag package-policy-2022-06

**Comment by leni-msft** [comment]
> > @robga To fix Avocado error, please make sure all APIs are included in the new default tag
> 
> @leni-msft I think that is a false positive error. All the APIs have been included in the new tag package-policy-2022-06

@ruowan please help check

**Comment by jianyexi** [comment]
the lintDiff exception is because there are different '$.securityDefinitions.azure_auth.description' under swagger file in the default tag:
package-policy-2022-06
``` yaml $(tag) == 'package-policy-2022-06'
input-file:
- Microsoft.Authorization/stable/2020-09-01/dataPolicyManifests.json
- Microsoft.Authorization/stable/2021-06-01/policyDefinitions.json
- Microsoft.Authorization/stable/2021-06-01/policySetDefinitions.json
- Microsoft.Authorization/stable/2022-06-01/policyAssignments.json
- Microsoft.Authorization/preview/2022-07-01-preview/policyExemptions.json
- Microsoft.Authorization/preview/2022-08-01-preview/policyVariables.json
- Microsoft.Authorization/preview/2022-08-01-preview/policyVariableValues.json
# Needed when there is more than one input file
override-info:
  title: PolicyClient
```

**Comment by robga** [comment]
> the lintDiff exception is because there are different '$.securityDefinitions.azure_auth.description' under swagger file in the default tag: package-policy-2022-06
> 
> ```yaml
> input-file:
> - Microsoft.Authorization/stable/2020-09-01/dataPolicyManifests.json
> - Microsoft.Authorization/stable/2021-06-01/policyDefinitions.json
> - Microsoft.Authorization/stable/2021-06-01/policySetDefinitions.json
> - Microsoft.Authorization/stable/2022-06-01/policyAssignments.json
> - Microsoft.Authorization/preview/2022-07-01-preview/policyExemptions.json
> - Microsoft.Authorization/preview/2022-08-01-preview/policyVariables.json
> - Microsoft.Authorization/preview/2022-08-01-preview/policyVariableValues.json
> # Needed when there is more than one input file
> override-info:
>   title: PolicyClient
> ```

@jianyexi  In the new tag package-policy-2022-06, I only added the following two new files:
- Microsoft.Authorization/stable/2022-06-01/policyAssignments.json
- Microsoft.Authorization/preview/2022-07-01-preview/policyExemptions.json

I fixed '$.securityDefinitions.azure_auth.description' error in these two new files. (Which is simply missing period at the end of the description) I don't want to touch the existing files because that will trigger more validation on them. I think the error in the existing files should be fixed by next time we update the API-Version for them. Also, I think the check should skip the errors from the file not changed in the PR.

**Comment by robga** [comment]
> * Resolve conflict readme file
> * This path /providers/Microsoft.Management/managementGroups/{}/providers/Microsoft.Authorization/policyAssignments is defined in resource-manager/Microsoft.Authorization/stable/2022-06-01/policyAssignments.json but was missing in default tag. Is this expected?
> * Pls fix conflict model for lintDiff
> 
> ```
> '$.securityDefinitions.azure_auth.description' has incompatible values (---Azure Active Directory OAuth2 Flow,
> ---Azure Active Directory OAuth2 Flow.).". 
> ```

@ruowan 
* I have resolved the conflict.
* When you say "it was missing in the default tag", where is this "default tag"? 
* In the new tag package-policy-2022-06, I only added the following two new files:

    - Microsoft.Authorization/stable/2022-06-01/policyAssignments.json
    - Microsoft.Authorization/preview/2022-07-01-preview/policyExemptions.json

I fixed '$.securityDefinitions.azure_auth.description' error in these two new files. (Which is simply missing period at the end of the description) I don't want to touch the existing files because that will trigger more validation on them. I think the error in the existing files should be fixed by next time we update the API-Version for them. Also, I think the check should skip the errors from the file not changed in the PR.

**Review by ruowan — APPROVED** [review]
